R/call_utils.R

```diff
@@ -49,108 +21,96 @@ call_check_parse_varname <- function(varname) {
 #'   }
 #'
 #' @examples
-#' teal.slice:::call_condition_choice("SEX", choices = c(1, 2))
-#' teal.slice:::call_condition_choice(as.name("SEX"), choices = "F")
-#' teal.slice:::call_condition_choice("SEX", choices = c("F", "M"))
-#' teal.slice:::call_condition_choice("SEX", choices = factor(c("F", "M")))
-#' teal.slice:::call_condition_choice("x$SEX", choices = Sys.Date())
-#' teal.slice:::call_condition_choice("trunc(x$SEX)", choices = Sys.time())
+#' \dontrun{
+#' call_condition_choice("SEX", choices = c(1, 2))
+#' call_condition_choice("SEX", choices = "F")
+#' call_condition_choice("SEX", choices = c("F", "M"))
+#' call_condition_choice("x$SEX", choices = Sys.Date())
+#' call_condition_choice("x$SEX", choices = Sys.time())
+#' }
 #' @return a `call`
 #' @keywords internal
 call_condition_choice <- function(varname, choices) {
-  varname <- call_check_parse_varname(varname)
-
-  if (is.factor(choices)) {
-    choices <- as.character(choices)
-  } else if (inherits(choices, "Date")) {
-    choices <- format(choices)
-  } else if (inherits(choices, c("POSIXct", "POSIXlt"))) {
-    choices <- format(choices)
-  }
-
+  checkmate::assert_string(varname)
+  checkmate::assert_multi_class(
+    choices,
+    c("integer", "numeric", "character", "factor", "Date", "POSIXct", "POSIXlt")
+  )
 
-  if (length(choices) == 1) {
-    call("==", varname, choices)
+  if (is.numeric(choices)) {
+    if (length(choices) == 1) {
+      sprintf("%s == %s", varname, sprintf("%s", choices))
```

Comment:
```suggestion
      sprintf("%s == %s", varname, choices)
```

`%s` causes conversion to string. Alternatively you can use `as.character(choices)`